crates/core/tedge_api/src/workflow/log/logged_command.rs

```diff
@@ -142,7 +143,10 @@ impl LoggedCommand {
     /// In contrast to [`std::process::Command`], `program` can contain space-separated arguments,
     /// which will be properly parsed, split, and passed into `.args()` call for the underlying
     /// command.
-    pub fn new(program: impl AsRef<OsStr>) -> Result<LoggedCommand, std::io::Error> {
+    pub fn new(
+        program: impl AsRef<OsStr>,
+        tmp_dir: impl AsRef<Path>,
```

Comment:
`working_dir` would be more appropriate.

```suggestion
        working_dir: impl AsRef<Path>,
```